libs/traits/src/lib.rs

```diff
@@ -187,6 +187,18 @@ pub trait PoolReserve<AccountId, CurrencyId>: PoolInspect<AccountId, CurrencyId>
 	fn deposit(pool_id: Self::PoolId, from: AccountId, amount: Self::Balance) -> DispatchResult;
 }
 
+/// A trait that supports modifications of pool write-off policies
+pub trait PoolWriteOffPolicyMutate<PoolId> {
+	type Rate;
+	type WriteOffRule: Encode + Decode + Clone + TypeInfo + Debug + PartialEq;
+	type MaxWriteOffPolicySize: Get<u32>;
+
+	fn update(
+		pool_id: PoolId,
+		policy: BoundedVec<Self::WriteOffRule, Self::MaxWriteOffPolicySize>,
```

Comment:
I feel like having a `BoundedVec` with a `MaxPolicySize` seems to be too much implementation detail of how it is currently handled from `pallet-loans-ref`. What do you think about having:

```rust
policy: impl IntoIterator<Item = Self::WriteOffRule>
```

In this way, any kind of list can be used as a policy, and it is the implementation that needs to ensure it works for the way it is using it.

Reply:
Since the data is coming in as a `BoundedVec` from an extrinsic call anyway, I think converting it to an iterator and then re-collecting on the other side isn't really worth the slightly nicer API. We can always clean it up later if we end up with another potential consumer of this trait.

Reply:
No strong opinion if now we have the same structure in both places 👍🏻

Reply:
I would keep it as is then :)